### PR TITLE
[interp] Remove usage of some gotos

### PR DESCRIPTION
**Pull request by BrzVlad**
Their usage resulted in worse code on wasm.

### CODE REVIEW

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -6986,7 +6986,7 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 		MINT_IN_CASE(MINT_LOCALLOC) {
 			stackval *sp_start = (stackval*)(locals + frame->imethod->total_locals_size + frame->imethod->vt_stack_size);
 			if (sp != sp_start + 1) /*FIX?*/
-				goto abort_label;
+				THROW_EX (mono_get_exception_execution_engine (NULL), ip); \
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:6989]: No need for trailing backslash.

File: src/mono/mono/mini/interp/interp.c [hunk 2]
```diff
@@ -5255,7 +5255,7 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 					if (isinst_instr)
 						sp [-1].data.p = NULL;
 					else
-						goto invalid_cast_label;
+						THROW_EX (mono_get_exception_invalid_cast (), ip);
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:5258]: Maybe a macro, `THROW_INVALD_CAST`?

File: src/mono/mono/mini/interp/interp.c [hunk 3]
```diff
@@ -5057,7 +5057,7 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			OBJREF (o) = mono_gc_alloc_obj (vtable, m_class_get_instance_size (vtable->klass));
 			if (G_UNLIKELY (!o)) {
 				mono_error_set_out_of_memory (error, "Could not allocate %i bytes", m_class_get_instance_size (vtable->klass));
-				goto throw_error_label;
+				THROW_EX (mono_error_convert_to_exception (error), ip);
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:5060]: Maybe a macro `THROW_ERROR`?

File: src/mono/mono/mini/interp/interp.c [hunk 4]
```diff
@@ -4714,12 +4714,12 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
...
 		MINT_IN_CASE(MINT_REM_UN_I4)
 			if (sp [-1].data.i == 0)
-				goto div_zero_label;
+				THROW_EX (mono_get_exception_divide_by_zero (), ip);
 			BINOP_CAST(i, %, guint32);
 			MINT_IN_BREAK;
 		MINT_IN_CASE(MINT_REM_UN_I8)
 			if (sp [-1].data.l == 0)
-				goto div_zero_label;
+				THROW_EX (mono_get_exception_divide_by_zero (), ip);
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:4722]: Maybe a macro `THROW_DIV_ZERO`?

File: src/mono/mono/mini/interp/interp.c [hunk 5]
```diff
@@ -4672,31 +4672,31 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
...
 		}
 		MINT_IN_CASE(MINT_REM_I8) {
 			gint64 l1 = sp [-1].data.l;
 			gint64 l2 = sp [-2].data.l;
 			if (l1 == 0)
-				goto div_zero_label;
+				THROW_EX (mono_get_exception_divide_by_zero (), ip);
 			if (l1 == (-1) && l2 == G_MININT64)
-				goto overflow_label;
+				THROW_EX (mono_get_exception_overflow (), ip);
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:4699]: Maybe a macro `THROW_OVERFLOW`?

File: src/mono/mono/mini/interp/interp.c [hunk 6]
```diff
@@ -1090,7 +1090,7 @@ interp_throw (ThreadContext *context, MonoException *ex, InterpFrame *frame, con
 
 #define NULL_CHECK(o) do { \
 	if (G_UNLIKELY (!(o))) \
-		goto null_label; \
+		THROW_EX (mono_get_exception_null_reference (), ip); \
```
Comment (jaykrell) [inline src/mono/mono/mini/interp/interp.c:1093]: Maybe a macro `THROW_NULL_DEREFERENCE`?

**Comment by jaykrell** [comment]
This should make the overall code a little larger and intermix more cold code into the hot paths, but if compiler's do better with it, oh well.

**Review by vargaz — APPROVED** [review]

**Pull request merged**